File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */

//  function to convert linkedlist to array
 function toArray(list){
    let container = [];
    while(list){
       container.push(list.val);
       list = list.next; 
    }
    return container;
 }

// funtion to convert array to linkedList
 function toList(arr){
    let newNode = new ListNode(0);
    let current = newNode;
    for(let i = 0; i < arr.length; i++){
        current.next = new ListNode(arr[i]);
        current = current.next;
    }
    return newNode.next;
 }
 
//  function to merge two lists
 function mergeTwoLists(list1, list2) {
    let arrList1 = toArray(list1);
    let arrList2 = toArray(list2);
    let sortedArr = [...arrList1,...arrList2];
    let res = sortedArr.sort((a,b)=>a-b);
    return toList(res);
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummyNode = ListNode() 
        current = dummyNode

        ptr1 = list1
        ptr2 = list2

        while ptr1 and ptr2:
            if ptr1.val <= ptr2.val:
                current.next = ptr1
                ptr1 = ptr1.next
            else:
                 current.next = ptr2
                 ptr2 = ptr2.next
            current = current.next
        if ptr1:
         current.next = ptr1
        elif ptr2:
         current.next = ptr2
        return dummyNode.next

